x-pack/plugins/lens/common/embeddable_factory/index.ts

```diff
@@ -17,7 +18,8 @@ export type LensEmbeddablePersistableState = EmbeddableStateWithType & {
 };
 
 export const inject: EmbeddableRegistryDefinition['inject'] = (state, references) => {
-  const typedState = state as LensEmbeddablePersistableState;
+  // We need to clone the state because we can not modify the original state object.
+  const typedState = cloneDeep(state) as LensEmbeddablePersistableState;
```

Comment:
@dej611 please make sure to include this change when working on embeddable refactor 👌🏼